### PR TITLE
JAMES-3210 add integration tests to verify that the bounce is sent to the sender

### PR DESCRIPTION
**Pull request by remk**
follow up to  #3435
rebased on #3441

### CODE REVIEW

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java [hunk 1]
```diff
@@ -86,6 +92,46 @@ public void dsnBounceMailetShouldDeliverBounce() throws Exception {
...
+    public void dsnBounceMailetShouldDeliverBounceToTheMailFromAddress() throws Exception {
+        setup(DSNBounce.class);
+
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+                .sendMessageWithHeaders(SENDER, RECIPIENT,
+                        "From: " + OTHER + "\r\n" +
+                                "To: " + RECIPIENT + "\r\n" +
+                                "Subject: " + "Hello\r\n" +
+                                "\r\n" +
+                                "Please bounce me to the return address\r\n"
```
Comment (Arsnael) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java:105]: indent

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java [hunk 2]
```diff
@@ -86,6 +92,46 @@ public void dsnBounceMailetShouldDeliverBounce() throws Exception {
...
+                );
+
+        testIMAPClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+                .login(SENDER, SENDER_PASSWORD)
+                .select(TestIMAPClient.INBOX)
+                .awaitMessage(awaitAtMostOneMinute);
```
Comment (Arsnael) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java:111]: I feel like in your last PRs I can see more and more indents with 8 spaces in your java code, while I believe the standard in James is 4? Maybe you need to change that config in your IDE?

**Comment by Arsnael** [comment]
```
[a7faec13ac0f791d46f9878f29caefc3d2813f7d] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project james-server-mailets-integration-testing: There was a timeout or other error in the fork -> [Help 1]
```
Not sure if related, not much more info unfortunately...

test this please

**Comment by chibenwa** [comment]
test this please (build rescheduled)

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java (hunk 2 above)
Comment (remk) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java:111]: I think i have updated it, but i will check it out :)

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java [hunk 3]
```diff
@@ -71,177 +81,158 @@ public void tearDown() {
...
+
+    @Test
+    public void dsnBounceMailetBouncedMailShouldBeAdressedToTheSenderInEnvelopeAndHeader() throws Exception {
+        setup(DSNBounce.class);
+
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+                .sendMessageWithHeaders(SENDER, RECIPIENT,
+                        "From: " + OTHER + "\r\n" +
+                                "To: " + RECIPIENT + "\r\n" +
+                                "Subject: " + "Hello\r\n" +
```
[the remark this answers is not in the document]
Comment (remk) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java]: i think i prefer the version with the line break as it is closer visually to the smtp message we want to send. But it's subjective.

**Comment by chibenwa** [comment]
404 test this please

**Comment by chibenwa** [comment]
```
[2c6a6b1125df02cc1842304873f874eda9cb5432] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project james-server-mailets-integration-testing: There was a timeout or other error in the fork -> [Help 1]
```

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**